File: torch3d/ops/voxelize.py
import torch


def voxelize_point_cloud(
    points,
    voxel_size,
    grid_size,
    min_range,
    max_range,
    max_points_per_voxels=128,
    max_voxels=20000,
):
    if grid_size is None:
        grid_size = torch.floor((max_range - min_range) / voxel_size)
    _C = _lazy_import()
    voxels, indices, num_points_per_voxel = _C.voxelize_point_cloud(
        points,
        voxel_size,
        grid_size,
        min_range,
        max_range,
        max_points_per_voxels,
        max_voxels,
    )
    return voxels, indices, num_points_per_voxel

File: torch3d/csrc/cpu/voxelize.cpp
#include "cpu.h"


template <typename T>
void voxelize_point_cloud_impl(const T* points)
{
}


std::vector<at::Tensor> voxelize_point_cloud_cpu(
    const at::Tensor& points,
    const at::Tensor& voxel_size,
    const at::Tensor& grid_size,
    const at::Tensor& min_range,
    const at::Tensor& max_range,
    int max_points_per_voxel,
    int max_voxels)
{
    int batch_size = points.size(0);
    int in_channels = points.size(1);
    int num_points = points.size(2);

    AT_DISPATCH_FLOATING_TYPES(points.scalar_type(), "voxelize_point_cloud_cpu", [&] {
        voxelize_point_cloud_impl<scalar_t>(points.data_ptr<scalar_t>());
    });
}
